frontend/app/src/components/app/AppCore.vue

```diff
@@ -122,10 +120,30 @@ export default defineComponent({
       setLanguage(language);
     });
 
+    const premium = getPremium();
+    const { fetchV3Balances: fetchUniswapV3Balances } = useUniswapStore();
+    const { fetchBalances: fetchSushiswapBalances, reset: resetSushiswap } =
+      useSushiswapStore();
+    const { fetchBalances: fetchBalancerBalances, reset: resetBalancer } =
+      useBalancerStore();
+
     watch(premium, (curr, prev) => {
-      const currentStatus = getStatus(defiUniswapV3Section);
-      if (prev !== curr && currentStatus !== Status.NONE) {
-        fetchV3Balances(true);
+      if (prev !== curr) {
+        if (getStatus(Section.DEFI_UNISWAP_V3_BALANCES) !== Status.NONE) {
+          fetchUniswapV3Balances(true);
+        }
+
+        if (curr) {
+          if (getStatus(Section.DEFI_SUSHISWAP_BALANCES) !== Status.NONE) {
+            fetchSushiswapBalances(true);
+          }
+          if (getStatus(Section.DEFI_BALANCER_BALANCES) !== Status.NONE) {
+            fetchBalancerBalances(true);
+          }
+        } else {
+          resetSushiswap();
+          resetBalancer();
+        }
```

Comment:
Wouldn't it make sense to not have this functionality in the AppCore component but into some store maybe 'defi/liquidity' or something.

Somehow it doesn't feel like the app-core should be aware of all the different protocols we support.